src/models/slashClient.ts

```diff
@@ -401,4 +460,78 @@ export class SlashClient {
 
     cmd.handler(interaction)
   }
+
+  async verifyKey(
+    rawBody: string | Uint8Array | Buffer,
+    signature: string,
+    timestamp: string
+  ): Promise<boolean> {
+    if (this.publicKey === undefined)
+      throw new Error('Public Key is not present')
+    return edverify(
+      signature,
+      Buffer.concat([
+        Buffer.from(timestamp, 'utf-8'),
+        Buffer.from(
+          rawBody instanceof Uint8Array
+            ? new TextDecoder().decode(rawBody)
+            : rawBody
+        )
+      ]),
+      this.publicKey
+    ).catch(() => false)
+  }
+
+  async verifyOpineRequest(req: ORequest): Promise<boolean> {
+    const signature = req.headers.get('x-signature-ed25519')
+    const timestamp = req.headers.get('x-signature-timestamp')
+    const contentLength = req.headers.get('content-length')
+
+    if (signature === null || timestamp === null || contentLength === null)
+      return false
+
+    const body = new Uint8Array(parseInt(contentLength))
+    await req.body.read(body)
+
+    const verified = await this.verifyKey(body, signature, timestamp)
+    if (!verified) return false
+
+    return true
+  }
+
+  /** Middleware to verify request in Opine framework. */
+  async verifyOpineMiddleware(
+    req: ORequest,
+    res: OResponse,
+    next: CallableFunction
+  ): Promise<any> {
+    const verified = await this.verifyOpineRequest(req)
+    if (!verified) return res.setStatus(401).end()
+
+    await next()
+    return true
+  }
+
+  // TODO: create verifyOakMiddleware too

```

Comment:
i think there need draft

Reply:
No, it's just an idea to add it. I'm not good with Oak so.

Reply:
> No, it's just an idea to add it. I'm not good with Oak so.

Ok